include/HTTP/Request.h

```diff
@@ -0,0 +1,218 @@
+/**
+ * @author krrle
+ * @brief 
+ * @m_version 0.1
+ * 
+ * @copyright Copyright (c) 2021
+ * 
+ */
+#pragma once
+#include <iostream>
+#include <utility>
+#include <map>
+#include <URI/Uri.h>
+
+namespace HTTP
+{   
+
+/**
+    * @brief Enumeration for HTTP methods
+    * 
+    */
+enum class Method{
+    GET = 0,
+    HEAD = 1,
+    POST = 2, 
+    PUT = 3, 
+    DELETE = 4, 
+    CONNECT = 5, 
+    OPTIONS = 6, 
+    TRACE = 7, 
+    PATCH = 8,
+    COUNT = 9
+};
+
+/**
+    * @brief Enumeration for HTTP versions
+    * 
+    */
+enum class Version {
+    HTTP09 = 0,
+    HTTP10 = 1,
+    HTTP11 = 2,
+    HTTP20 = 3,
+};
+
+/**
+    * @brief Maping string with enumeration HTTP method
+    * 
+    */
+const std::map<std::string, Method> table = { 
+    { "GET",     Method::GET     }, 
+    { "HEAD",    Method::HEAD    },
+    { "POST",    Method::POST    }, 
+    { "PUT",     Method::PUT     }, 
+    { "DELETE",  Method::DELETE  }, 
+    { "CONNECT", Method::CONNECT },
+    { "OPTIONS", Method::OPTIONS },
+    { "TRACE",   Method::TRACE   },
+    { "PATCH",   Method::PATCH   },
+    { "COUNT",   Method::COUNT   }
+};
+
+/**
+    * @brief Maping string with enumeration HTTP-versions
+    * 
+    */
+const std::map<std::string, Version> table_versions = {
+    { "HTTP/0.9", Version::HTTP09 },
+    { "HTTP/1.0", Version::HTTP10 },
+    { "HTTP/1.1", Version::HTTP11 },
+    { "HTTP/2.0", Version::HTTP20 }
+};
+
+class Request{
+
+    public:
+
+    /**
+        * @brief Default constructor
+        * 
+        */
+    Request();
+
+    /**
+        * @brief Default destructor
+        * 
+        */
+    ~Request();
+
+    Request(const Request&) = delete;
+    Request& operator=(const Request&) = delete;
+    Request(Request&&) = delete;
+    Request& operator=(Request &&) = delete;
+
+    /**
+    * @brief Get the Method object
+    * 
+    * @return Method 
+    */
+    Method get_method();
+
+    /**
+        * @brief Get the version object
+        * 
+        * @return Version 
+        */
+    Version get_version();
+
+    /**
+        * @brief Parsing request line, if request line is valid returns true, otherwise returns false
```

Comment:
"if request line is valid returns true, otherwise returns false" - this should be comment for the return value